docs(Text): add description (#9583)
patternfly/patternfly-react: packages/react-core/src/components/Text/examples/Text.md

propComponents: ['TextContent', 'Text', 'TextList', 'TextListItem']
---

> The `Text` component provides simple, built-in styling for putting common blocks of HTML elements together. It establishes the block of content and styling within it for the elements listed in the `component` property(`h1` through `h6`, `p`, `a`, `small`, `blockquote` and `pre`), as well as the text component suite (`TextContent`, `TextList`, `TextListItem`).

Reviewer: ```suggestion
The `Text` component provides simple, built-in styling for putting common blocks of HTML elements together. It establishes the block of content and styling within it for the elements listed in the `component` property(`h1` through `h6`, `p`, `a`, `small`, `blockquote`, and `pre`), as well as the text component suite (`TextContent`, `TextList`, `TextListItem`).
```

Author: Funny cause I'm usually pro oxford comma, must've missed it haha.

> The `Text` component provides simple, built-in styling for putting common blocks of HTML elements together. It establishes the block of content and styling within it for the elements listed in the `component` property(`h1` through `h6`, `p`, `a`, `small`, `blockquote` and `pre`), as well as the text component suite (`TextContent`, `TextList`, `TextListItem`).

Reviewer 2: ```suggestion
The `<Text>` component provides simple, built-in styling for putting common blocks of HTML elements together. It establishes the block of content and styling within it for the elements listed in the `component` property (`h1` through `h6`, `p`, `a`, `small`, `blockquote` and `pre`), as well as the text component suite (`TextContent`, `TextList`, `TextListItem`).
```

adding onto what Michael said - just some formatting fixes

Reviewer: I like wrapping the component in `<>`, it helps differentiate a component from a prop. Tho should we also update this?

> (`TextContent`, `TextList`, `TextListItem`)

to 

> (`<TextContent>`, `<TextList>`, `<TextListItem>`).

Reviewer 2: oh yes, missed those!

> Other components are not supported as children of `Text`, and may cause styling overrides or other conflicts. For example, the `List` and `Title` components are meant for more complex use cases and should not be nested inside `Text` for simple blocks - `TextList` and `Text` with the `component="h1"` property should be used instead. Instead of `Divider`, `Text` expects to contain an `hr` which it styles as a divider.

Reviewer: 👍

> Other components are not supported as children of `Text`, and may cause styling overrides or other conflicts. For example, the `List` and `Title` components are meant for more complex use cases and should not be nested inside `Text` for simple blocks - `TextList` and `Text` with the `component="h1"` property should be used instead. Instead of `Divider`, `Text` expects to contain an `hr` which it styles as a divider.

Reviewer 2: ```suggestion
You cannot nest other components within `<Text>`, and doing so can cause styling overrides or other conflicts. Instead, you can use the `<Text>` component's properties to achieve the same results.  

For example, rather than nesting the `<List>` and `<Title>` components within `<Text>`, you should pass `component="h1"` into the `<TextList>` and `<Text>` components. Similarly, when you need to add a divider , rather than passing in a separate `<Divider>` component, you should utilize the `hr` property that `<Text>` supports, which will be styled as a divider.
```

How do you feel about this rewording? Just trying to lead with the information that is most important to get across, and then follow with the examples.  

Also, in the past I've style component names this way when I'm referring to the React component object, but if that's unnecessary we could change that strategy going forward.
